### PR TITLE
:seedling: Add probe registration mechanism

### PR DESCRIPTION
**Pull request by spencerschrock**
#### What kind of change does this PR introduce?

refactor

- [X] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
A list of probes and their mappings (name to runner func, which raw data it needs, ect) are maintained away from the probe code. This makes is easy for people to forget to update 

#### What is the new behavior (if this is a feature change)?**
The probes define it in their init block. Unfortunately, due to the fact that `init` is only executed when something imports it, we need to add the probes in the `entries.go` file anyway.

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
NONE

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

**Comment by spencerschrock** [comment]
Leaving as a draft to revisit tomorrow one last time. Not sure if this is noticably better than what we have now.
I think it's slightly more structured and defined closer to the probe which is nice, but setting the raw data is still a little messy

File: probes/codeReviewOneReviewers/impl.go [hunk 1]
```diff
@@ -22,23 +22,35 @@ import (
...
+func init() {
+	err := probes.Register(probes.Probe{
+		Name:            Probe,
+		Implementation:  Run,
+		RequiredRawData: []probes.CheckName{probes.CodeReview},
+	})
+	if err != nil {
+		panic(err)
+	}
+}
```
Comment (pnacht) [inline probes/codeReviewOneReviewers/impl.go:38]: This block is effectively identical across all probes. Is it possible to refactor this into:

```go
// probes/util.go or internal/probes.go
func ProbeInit(probe probes.Probe, impl func(raw *checker.RawResults), requiredRawData []probes.CheckName) {
    err := probes.Register(probes.Probe{
		Name:            probe,
		Implementation:  impl,
		RequiredRawData: requiredRawData,
	})
	if err != nil {
		panic(err)
	}
}

// $PROBE/impl.go
func init() {
    probes.ProbeInit(Probe, Run, []probes.CheckName{probes.CodeReview})
}
```

Comment (spencerschrock) [reply probes/codeReviewOneReviewers/impl.go]: Good suggestion. Done in 2ff7302b691d8964b5b0b6088abbdb707d8823cd.

**Review by laurentsimon — APPROVED** [review]
LGTM, thanks!

**Pull request merged**